ts/src/base/Exchange.ts

```diff
@@ -874,58 +877,117 @@ export default class Exchange {
+                this.nodeHttpModuleLoaded = true;
+                const httpsModule = await import (/* webpackIgnore: true */'node:https')
+                this.httpsAgent = new httpsModule.Agent ({ keepAlive: true });
+            }
+        }
 
         // ##### PROXY & HEADERS #####
         headers = this.extend (this.headers, headers);
-        const [ proxyUrl, httpProxy, httpsProxy, socksProxy ] = this.checkProxySettings (url, method, headers, body);
+        // proxy-url
+        const proxyUrl = this.checkProxyUrlSettings (url, method, headers, body);
         if (proxyUrl !== undefined) {
             // in node we need to set header to *
             if (isNode) {
-                headers = this.extend ({ 'Origin': this.origin }, headers)
+                headers = this.extend ({ 'Origin': this.origin }, headers);
+                if (proxyUrl.substring(0, 5) === 'https') {
+                    this.agent = this.httpsAgent;
+                } else {
+                    // for `http://` protocol proxy-urls, we need to load `http` module only on first call
+                    if (!this.httpAgent) {
+                        const httpModule = await import (/* webpackIgnore: true */'node:http')
+                        this.httpAgent = new httpModule.Agent ();
+                    }
+                    this.agent = this.httpAgent;
+                }
             }
             url = proxyUrl + url;
-        } else if (httpProxy !== undefined) {
-            const module = await import (/* webpackIgnore: true */ '../static_dependencies/proxies/http-proxy-agent/index.js')
-            const proxyAgent = new module.HttpProxyAgent(httpProxy);
-            this.agent = proxyAgent;
-        }  else if (httpsProxy !== undefined) {
-            const module = await import (/* webpackIgnore: true */ '../static_dependencies/proxies/https-proxy-agent/index.js')
-            const proxyAgent = new module.HttpsProxyAgent(httpsProxy);
-            proxyAgent.keepAlive = true;
-            this.agent = proxyAgent;
-        } else if (socksProxy !== undefined) {
-            let module = undefined;
-            try {
-                // @ts-ignore
-                module = await import (/* webpackIgnore: true */ 'socks-proxy-agent');
-            } catch (e) {
-                throw new NotSupported (this.id + ' - to use SOCKS proxy with ccxt, at first you need install module "npm i socks-proxy-agent" ');
-            }
-            this.agent = new module.SocksProxyAgent(socksProxy);
         }
-
+        // proxy agents
+        const [ httpProxy, httpsProxy, socksProxy ] = this.checkProxySettings (url, method, headers, body);
+        this.checkConflictingProxies (httpProxy || httpsProxy || socksProxy, proxyUrl);
+        await this.loadProxyModules (); // this is needed in JS, independently whether proxy properties were set or not, we have to load them because of necessity in WS, which would happen beyond 'fetch' method
```

Comment:
question, do we need to load them on every fetch call? or would once be enough?

Reply:
u'r right, just look into `loadProxyModules` method and you'll see that it's actually done only once. (btw, I'll simplify that)